ginga/tests/test_dask.py

```diff
@@ -0,0 +1,97 @@
+import hashlib
+import numpy as np
+import pytest
+
+da = pytest.importorskip('dask.array')
+
+from ginga import AstroImage, trcalc
+from ginga.misc import log
+
+
+class TestDask:
+    def setup_class(self):
+        self.logger = log.get_logger("TestDask", null=True)
+
+    def _2ddata(self, shape, data_np=None):
+        if data_np is None:
+            data_np = np.asarray([min(i, j)
+                                  for i in range(shape[0])
+                                  for j in range(shape[1])])
+        data_np = data_np.reshape(shape)
+        data_dk = da.from_array(data_np, chunks=100)
+        return data_dk
+
+    def _3ddata(self, shape, data_np=None):
+        if data_np is None:
+            data_np = np.asarray([min(i, j, k)
+                                  for i in range(shape[0])
+                                  for j in range(shape[1])
+                                  for k in range(shape[2])])
+        data_np = data_np.reshape(shape)
+        data_dk = da.from_array(data_np, chunks=100)
+        return data_dk
+
+    def test_dask_slice_trcalc(self):
+        """Test that we can get a subslice of a dask array.
+        """
+        arr_dk = self._2ddata((1000, 500))
+
+        x_slice, y_slice = slice(12, 499, 3), slice(10, 951, 11)
+        view = (y_slice, x_slice)
+        data_dk = trcalc.fancy_index(arr_dk, view)
+        # type is preserved for slicing dask arrays
+        assert isinstance(data_dk, da.core.Array)
+        assert data_dk.shape == (86, 163)
+        data_np = data_dk.compute()
+        assert isinstance(data_np[0, 0], np.integer)
+        res = '177e1ed261ea24df277511078631ec0f95dfc3e781ac15b2d200f0f0040282ae'
+        m = hashlib.sha256()
+        m.update(str(data_np.tolist()).encode())
+        assert m.hexdigest() == res
```

Comment:
Wow, I didn't realize the `dask` data comparison would require a hash! 👀

Reply:
Doesn't require it, but it makes the test for correct result simpler to code, and can basically reuse the formula with different hash values.